tests/system/src/test/java/test/javafx/scene/web/CSSRoundingTest.java

```diff
@@ -0,0 +1,173 @@
+            Scene scene = new Scene(webView, 150, 100);
+            cssRoundingTestApp.primaryStage.setScene(scene);
+            cssRoundingTestApp.primaryStage.show();
+        });
+    }
+
+    @Test public void testCSSroundingForLinearLayout() {
+
+        final CountDownLatch webViewStateLatch = new CountDownLatch(1);
+
+        Util.runAndWait(() -> {
+            assertNotNull(webView);
+
+            webView.getEngine().getLoadWorker().stateProperty().
+                    addListener((observable, oldValue, newValue) -> {
+                        if (newValue == SUCCEEDED) {
+                            webView.requestFocus();
+                        }
+                    });
+
+            webView.focusedProperty().
+                    addListener((observable, oldValue, newValue) -> {
+                        if (newValue) {
+                            webViewStateLatch.countDown();
+                        }
+                    });
+            String content = """
+                <html>
+                <head>
+                <style type="text/css">
+                    body, div {
+                        margin: 0;
+                        padding: 0;
+                        border: 0;
+                    }
+                    #top, #bottom {
+                        line-height: 1.5;
+                        font-size: 70%;
+                        background:green;
+                        color:white;
+                        width:100%;
+                    }
+                    #top {
+                        padding:.6em 0 .7em;
+                    }
+                    #bottom {
+                      position:absolute;
+                      top:2.8em;
+                    }
+                </style>
+                </head>
+                <body>
+                <div id="top">no gap below</div>
+                <div id="bottom">no gap above</div>
+                <div id="description"></div>
+                <div id="console"></div>
+                <script>
+                description("This test checks that floating point rounding doesn't cause misalignment.  There should be no gap between the divs.");
+                var divtop = document.getElementById("top").getBoundingClientRect();
+                var divbottom = document.getElementById("bottom").getBoundingClientRect();
+                console.log("divtop.bottom: " + divtop.bottom);
+                console.log("divbottom.top: " + divbottom.top);
+                window.testResults = { topBottom: Math.round(divtop.bottom), bottomTop: Math.round(divbottom.top) };
+                </script>
+                </body>
+                </html>
+                """;
+            webView.getEngine().loadContent(content);
+        });
+
+        assertTrue("Timeout when waiting for focus change ", Util.await(webViewStateLatch));
+        //introduce sleep , so that web contents would be loaded , then take snapshot for testing
+        Util.sleep(1000);
```

Comment:
The sleep may or may not be needed, but OK to leave it. The comment is wrong, though, since you aren't taking a snapshot.